packages/cloudflare/src/api/cloudflare-context.ts

```diff
@@ -53,26 +61,46 @@ type InternalGlobalThis<
 export function getCloudflareContext<
   CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
   Context = ExecutionContext,
->(): CloudflareContext<CfProperties, Context> {
+>(options: { async: true }): Promise<CloudflareContext<CfProperties, Context>>;
+export function getCloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+>(options?: { async?: false }): CloudflareContext<CfProperties, Context>;
+export function getCloudflareContext<
```

Comment:
I'm not really sure to like `getCloudflareContext({async:true})` over `getClouflfareContextAsync()` - I need to think more

Reply:
Yes I agree, I might also prefer to have a distinct function for the async version (it would certainly simplify the function's type) I went with this version since that's what was discussed and agreed on in our weekly meeting

I'm happy to have two separate functions if you prefer that, it does feel clearer to me

although there are also some downsides:
 - more overhead to developers as they need to be aware of two functions
 - more overhead on us to document both functions
 
 up to you, I am fine with either 👍

Reply:
I'm not sure, it's more a gut feeling but changing the return type according to the what parameter is passed does not look super clean to me.

> more overhead on us to document both functions

IDK, I feel like documenting `getCloudflareContext()` and `getCloudflareContext({async: true})` or `getCloudflareContext()` and `getCloudflareContextAsync()` is not really different?

Reply:
> I'm not sure, it's more a gut feeling but changing the return type according to the what parameter is passed does not look super clean to me.

yes, I agree

> IDK, I feel like documenting getCloudflareContext() and getCloudflareContext({async: true}) or getCloudflareContext() and getCloudflareContextAsync() is not really different?

the way I see it is that if you only have a single function you only need to document that and can as part of that mention that it has an "async mode", if you have two functions it probably can get a bit more verbose (not a huge deal either way)

Reply:
No strong pref, pick what your prefer